### PR TITLE
Implement Cluster Email Settings Resource

### PR DESCRIPTION
**Pull request by NavisJ**
# Description
Implement Cluster Email Settings Resource

# ISSUE TYPE
- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
Cluster Email

##### OUTPUT

image
image

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by NavisJ** [comment]
run e2e test

**Comment by NavisJ** [comment]
run e2e test

File: powerscale/provider/cluster_email_resource.go [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "ID of the Cluster Email Settings.",
+				MarkdownDescription: "ID of the Cluster Email Settings.",
+				Computed:            true,
+			},
+			"settings": schema.SingleNestedAttribute{
+				Description:         "Cluster email notification settings.",
+				MarkdownDescription: "Cluster email notification settings.",
+				Required:            true,
```
Comment (taohe1012) [inline powerscale/provider/cluster_email_resource.go:80]: I remember that for single resource, better use optional not required.

File: powerscale/provider/cluster_email_resource.go [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+							stringvalidator.OneOf("all", "severity", "category", "none"),
+						},
+					},
+					"mail_relay": schema.StringAttribute{
+						Description:         "The address of the SMTP server to be used for relaying the notification messages.  An SMTP server is required in order to send notifications.  If this string is empty, no emails will be sent.",
+						MarkdownDescription: "The address of the SMTP server to be used for relaying the notification messages.  An SMTP server is required in order to send notifications.  If this string is empty, no emails will be sent.",
+						Optional:            true,
+						Computed:            true,
+						Validators: []validator.String{
+							stringvalidator.RegexMatches(regexp.MustCompile(`(^$|^((([a-zA-Z0-9_][a-zA-Z0-9-]{0,61})?[a-zA-Z0-9])(\.([a-zA-Z0-9_][a-zA-Z0-9-]{0,61})?[a-zA-Z0-9])*)$|^([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])(\.([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])){3}$|^[0-9A-Fa-f]{1,4}(:[0-9A-Fa-f]{1,4}){0,5}::([0-9A-Fa-f]{1,4}(:[0-9A-Fa-f]{1,4}){0,5})?$|^[0-9A-Fa-f]{1,4}(:[0-9A-Fa-f]{1,4}){7}$)`), "Invalid SMTP server address"),
```
Comment (taohe1012) [inline powerscale/provider/cluster_email_resource.go:97]: for the complex regex validation, thinking better let onefs to check its value

**Review by RayLiu7 — APPROVED** [review]

File: powerscale/provider/cluster_email_resource.go (hunk 1 above)
Comment (NavisJ) [reply powerscale/provider/cluster_email_resource.go:80]: Thinking that for the "setting" resource, empty body block makes no sense. Whatever provided in the tf file should be synchronized with the powerscale side. As long as user wants to manage the cluster email resource, this attribute is needed. Besides, all the attributes of cluster email are included in this `settings` nested attribute, so make it required for better usage.

File: powerscale/provider/cluster_email_resource.go (hunk 2 above)
Comment (NavisJ) [reply powerscale/provider/cluster_email_resource.go:97]: OK. will remove the regex check

**Comment by NavisJ** [comment]
run e2e test

**Review by taohe1012 — APPROVED** [review]

**Pull request merged**